defi/src/api2/cron-task/dimensions.ts

```diff
@@ -63,14 +63,402 @@ const timeData = {
+    m['dr'] = {
+      monthly: diffMaps(df.monthly, _dssr.monthly),
+      quarterly: diffMaps(df.quarterly, _dssr.quarterly),
+    }
+  }
+  // dhr = dr - dssr
+  if (!dhr && (m['dr'] as MQ) && (m['dssr'] as MQ)) {
+    const _dr = m['dr'] as MQ
+    const _dssr = m['dssr'] as MQ
+    m['dhr'] = {
+      monthly: diffMaps(_dr.monthly,   _dssr.monthly),
+      quarterly: diffMaps(_dr.quarterly, _dssr.quarterly),
+    }
+  }
+
+  // Labels (bl)
+  const dfbl: MQByLabel | undefined = m['dfbl'] as MQByLabel
+  const drbl: MQByLabel | undefined = m['drbl'] as MQByLabel
+  const dssrbl: MQByLabel | undefined = m['dssrbl'] as MQByLabel
+  const dhrbl: MQByLabel | undefined = m['dhrbl'] as MQByLabel
+
+  function diffNumberMaps(a?: Record<string, number>, b?: Record<string, number>) {
+    const out: Record<string, number> = {}
+    const keys = new Set([...(a ? Object.keys(a) : []), ...(b ? Object.keys(b) : [])])
+    keys.forEach(k => out[k] = (a?.[k] ?? 0) - (b?.[k] ?? 0))
+    return out
+  }
+
+  function diffLabelMaps(a?: Record<string, Record<string, number>>, b?: Record<string, Record<string, number>>) {
+    const out: Record<string, Record<string, number>> = {}
+    const labels = new Set([...(a ? Object.keys(a) : []), ...(b ? Object.keys(b) : [])])
+    labels.forEach(label => {
+      out[label] = diffNumberMaps(a?.[label], b?.[label])
+    })
+    return out
+  }
+
+  if (!dssrbl && dfbl && drbl) {
+    m['dssrbl'] = {
+      monthly: diffLabelMaps(dfbl.monthly, drbl.monthly),
+      quarterly: diffLabelMaps(dfbl.quarterly, drbl.quarterly),
+    }
+  }
+
+  if (!drbl && dfbl && m['dssrbl']) {
+    const _dssrbl = m['dssrbl'] as MQByLabel
+    m['drbl'] = {
+      monthly: diffLabelMaps(dfbl.monthly, _dssrbl.monthly),
+      quarterly: diffLabelMaps(dfbl.quarterly, _dssrbl.quarterly),
+    }
+  }
+
+  if (!dhrbl && (m['drbl'] as MQByLabel) && (m['dssrbl'] as MQByLabel)) {
+    const _drbl = m['drbl'] as MQByLabel
+    const _dssrbl = m['dssrbl'] as MQByLabel
+    m['dhrbl'] = {
+      monthly: diffLabelMaps(_drbl.monthly, _dssrbl.monthly),
+      quarterly: diffLabelMaps(_drbl.quarterly, _dssrbl.quarterly),
+    }
+  }
+}
+
+function serializeAggregate(proto: AggregatedProtocolOut & { generatedAt?: number }) {
+  const { adapterType, slug, metrics } = proto
+  const meta = proto.metadata || {}
+  return {
+    adapterType,
+    slug,
+    metrics,
+    ...meta, // flatten metadata at root
+  }
+}
+
+async function generateAggregatesFromBuild() {
+  const buildDir = findBuildDir()
+  if (!buildDir) return
+  const dimRoot = path.join(buildDir, 'dimensions')
+  if (!fs.existsSync(dimRoot)) return
+
+  // FEES only
+  const adapterTypes = [AdapterType.FEES]
+
+  for (const adapterType of adapterTypes) {
+    const adapterPath = path.join(dimRoot, adapterType)
+    if (!fs.existsSync(adapterPath)) continue
+
+    const recordTypeDirs = fs
+      .readdirSync(adapterPath)
+      .filter((d) => /[a-z]+protocol$/i.test(d))
```

Comment:
tbh, I really dont like this flow. 

Probably good seperation of code, makes it easier to read, disable only aggregated part, but more efficient to do it in one place instead of writing to a file and then reading it. We can use a flag like `generateBreakdown` to enable/disable certain code block